baggage-processor/src/main/java/io/opentelemetry/contrib/baggage/processor/BaggageSpanProcessor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.baggage.processor;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/**
+ * This span processor copies attributes stored in {@link Baggage} into each newly created {@link
+ * io.opentelemetry.api.trace.Span}.
+ */
+public class BaggageSpanProcessor implements SpanProcessor {
+  @Override
+  public void onStart(Context parentContext, ReadWriteSpan span) {
+    Baggage.fromContext(parentContext)
+        .forEach((s, baggageEntry) -> span.setAttribute(s, baggageEntry.getValue()));
```

Comment:
Consider allowing a subset of baggage entries instead via an optional constructor argument that defines a filter.

Reply:
Hey @tylerbenson - I've opened the following issue to track adding a user-defined function to determine if an attribute should be copied. I've been discussing options with a few contrib SIGs as a way to limit sensitive data being copied.
- https://github.com/open-telemetry/opentelemetry-java-contrib/issues/1295

The user-defined function is the most flexible as it can be configured as a allow or deny list using prefixes or whole matches.

The main concern with that approach is how it could be used with the Java Agent. An allow or deny list provided by system settings or env variables would be simplest solution but it's open to discussion. I'm aiming for consistency across language implementations so decided to add the processor without that feature (plus clear docs) while gathering feedback from each SIG.

Reply:
in the Java SIG, we discussed how to integrate this component into the javaagent.

We decided that the configuration (what baggage items to include) can and should be done here in the contrib repo as well, using an [AutoConfigurationCustomizer](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java)

I'm happy to work with you together on this part - as a follow up PR :smile: 

See https://docs.google.com/document/d/1WK9h4p55p8ZjPkxO75-ApI9m0wfea6ENZmMoFRvXSCw/edit#bookmark=id.lm2raghsdnup

Reply:
That's great to hear @zeitlinger - thank you for reporting back.

Yep, happy to work with you on making this processor accessible in the java agent 👍🏻

Reply:
Hi! Have the work on adding this to the agent already started somewhere ? I have found https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4520 issue that was opened a while ago before this contribution, but no follow-up PR to implement it. I'd be happy to help making progress here.

Reply:
hey @SylvainJuge! check out #1330

Reply:
I've just opened https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11697 to add it to the agent.